### PR TITLE
Fix problems when overriding the hand

### PR DESCRIPTION
**Pull request by Ekdohibs**
- If the hand can dig a node the item wielded can't, allow to dig it anyway.
- Fix the API callbacks from setting the hand instead of the wielded item.

See #5200 as well.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -3397,13 +3397,11 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
...
-	if (playeritem.getDefinition(itemdef_manager).name.empty()) { // override the hand
-		InventoryList *hlist = local_inventory->getList("hand");
-		if (hlist)
-			playeritem = hlist->getItem(0);
-	}
 	const ItemDefinition &playeritem_def =
 			playeritem.getDefinition(itemdef_manager);
+	InventoryList *hlist = local_inventory->getList("hand");
+	const ItemDefinition &hand_def =
+		hlist?hlist->getItem(0).getDefinition(itemdef_manager):itemdef_manager->get("");
```
Comment (Zeno-) [inline src/game.cpp]: It's much easier to read if there are spaces around '?' and ':'

File: src/game.cpp [hunk 2]
```diff
@@ -3802,7 +3810,9 @@ void Game::handleDigging(const PointedThing &pointed, const v3s16 &nodepos,
 
 	// If can't dig, try hand
 	if (!params.diggable) {
-		const ItemDefinition &hand = itemdef_manager->get("");
+		InventoryList *hlist = local_inventory->getList("hand");
+		const ItemDefinition &hand =
+			hlist?hlist->getItem(0).getDefinition(itemdef_manager):itemdef_manager->get("");
```
Comment (Zeno-) [inline src/game.cpp]: ?: operator formatting

File: src/network/serverpackethandler.cpp [hunk 3]
```diff
@@ -1375,7 +1375,10 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		const ItemDefinition &playeritem_def =
 			playersao->getWieldedItem().getDefinition(m_itemdef);
 		float max_d = BS * playeritem_def.range;
-		float max_d_hand = BS * m_itemdef->get("").range;
+		InventoryList *hlist = playersao->getInventory()->getList("hand");
+		const ItemDefinition &hand_def =
+			hlist?(hlist->getItem(0).getDefinition(m_itemdef)):(m_itemdef->get(""));
```
Comment (Zeno-) [inline src/network/serverpackethandler.cpp]: ?: operator

File: src/network/serverpackethandler.cpp [hunk 4]
```diff
@@ -1503,15 +1506,17 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
...
 						playeritem.getToolCapabilities(m_itemdef);
 				// Get diggability and expected digging time
 				DigParams params = getDigParams(m_nodedef->get(n).groups,
 						&playeritem_toolcap);
 				// If can't dig, try hand
 				if (!params.diggable) {
-					const ItemDefinition &hand = m_itemdef->get("");
+					InventoryList *hlist = playersao->getInventory()->getList("hand");
+					const ItemDefinition &hand =
+						hlist?hlist->getItem(0).getDefinition(m_itemdef):m_itemdef->get("");
```
Comment (Zeno-) [inline src/network/serverpackethandler.cpp]: ?: operator

**Comment by Zeno-** [comment]
Haven't tested, but LGTM

File: src/game.cpp (hunk 1 above)
Comment (paramat) [inline src/game.cpp]: Agreed.

**Comment by sofar** [comment]
Confirmed fixes #5200 with the code posted in that bug report.

**Comment by sofar** [comment]
Counting @Zeno- 's LGTM as an approval as well.

:+1:

**Pull request merged**